apps/web/src/utils/kv/kv.ts

```diff
@@ -0,0 +1,175 @@
+import Redis, { type Redis as RedisType } from 'ioredis';
+import { isDevelopment } from 'apps/web/src/constants';
+import { logger } from '../logger';
+
+type KvConstructorParam =
+  | { url: string; tls?: boolean }
+  | { host: string; port: number; username?: string; password?: string; tls?: boolean };
+
+/**
+ * Provides a limited, type-safe interface to Redis operations.
+ * Intentionally restricts access to dangerous commands and raw client operations.
+ */
+export class KVManager {
+  private client: RedisType | null = null;
+
+  private readonly connectionArg: KvConstructorParam;
+
+  private readonly connectionTls: boolean;
+
+  constructor(connectionParam: KvConstructorParam) {
+    if (!connectionParam || (!('url' in connectionParam) && !('host' in connectionParam))) {
+      throw new Error('No URL or options provided to KVManager');
+    }
+    this.connectionArg = connectionParam;
+    this.connectionTls = connectionParam.tls ?? false;
+  }
+
+  private async getClient(): Promise<RedisType> {
+    if (!this.client) {
+      console.log(
+        'creating new redis client: ',
+        'url' in this.connectionArg ? this.connectionArg.url : this.connectionArg.host,
+      );
+      if (!this.connectionArg) {
+        throw new Error('No URL or options provided to KVManager');
+      }
+
+      if ('url' in this.connectionArg) {
+        this.client = new Redis(this.connectionArg.url, this.connectionTls ? { tls: {} } : {});
+      } else {
+        this.client = new Redis({
+          ...this.connectionArg,
+          tls: this.connectionTls ? {} : undefined,
+        });
+      }
+
+      console.log('redis client created', this.client);
+    }
+
+    try {
+      console.log(`pinging ${this.connectionArg}`);
+      const pingRes = await this.client.ping();
+      console.log('ping response', pingRes);
+    } catch (err) {
+      if (!isDevelopment) {
+        logger.error('KV connection failed', err);
+      }
+      console.error(err);
+      throw new Error(`Failed to connect to KV: ${err}`);
+    }
+
+    return this.client;
+  }
+
+  async ping() {
+    if (this.client) {
+      try {
+        const pingRes = await this.client.ping();
+        return pingRes;
```

Comment:
```suggestion
        return this.client.ping();
```

Reply:
should keep the await in here right? 

`return await this.client.ping();` no?